script.rb

```diff
@@ -0,0 +1,81 @@
+module Enumerable
+  def my_each(array)
+    index = 0
+    while index < array.length
+      yield(array[index])
+      index += 1
+    end
+    array
+  end
+
+  def my_each_with_index(array)
+    index = 0
+    while index < array.length
+      yield(array[index], index)
+      index += 1
+    end
+    array
+  end
+
+  def my_select(array)
+    new_arr = []
+    my_each(array) do |x|
+      new_arr.push(x) if yield(x) == true
+    end
+    new_arr
+  end
+
+  def my_all?(array)
+    p my_select(array) { |x| yield(x) }.length == array.length
+  end
+
+  def my_any?(array)
+    p my_select(array) { |x| yield(x) }.length.positive? ? true : false
+  end
```

Comment:
- [ ] should have working `my_any?` identical and returns the same thing as ruby's [`any?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-any-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_any.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

 #### my_any? 
> Remember to test it not only for arrays but for ranges as well

- [ ] when no block or argument is given returns true if at least one of the collection is not false or nil

```ruby     
[nil, false, true, []].my_any? == [nil, false, true, []].any? => true
```
- Yours returns an ArgumentError
![image](https://user-images.githubusercontent.com/52209067/77689397-31709d80-6f67-11ea-8762-a2c98c02214c.png)

- Real any?
![image](https://user-images.githubusercontent.com/52209067/77689425-3f262300-6f67-11ea-88a9-2885ebab08d4.png)

- [ ] when a class is passed as an argument returns true if at least one of the collection is a member of such class

```ruby
[1, 2.5, 'a', 9].my_any?(Integer) == [1, 2.5, 'a', 9].any?(Integer) => true
```
- Yours is retrieving a NoMethodError
![image](https://user-images.githubusercontent.com/52209067/77689571-7dbbdd80-6f67-11ea-9d07-6c81c9bc2915.png)

- Real any?
![image](https://user-images.githubusercontent.com/52209067/77689595-8f04ea00-6f67-11ea-9f51-4536c3f8049b.png)

- [ ] when a Regex is passed as an argument returns false if none of the collection matches the Regex

```ruby
%w[dog door rod blade].my_any?(/z/) == %w[dog door rod blade].any?(/z/) => false
```
- Yours retrieves a NoMethodError
![image](https://user-images.githubusercontent.com/52209067/77689711-beb3f200-6f67-11ea-934a-7a0a918e0455.png)

- Real any?
![image](https://user-images.githubusercontent.com/52209067/77689742-cf646800-6f67-11ea-9e3e-eec7a960fe38.png)

- [ ] when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern

```ruby
[3,4,7,11].my_any?(3) should be equal to this [3,4,7,11].any?(3) => true
```
- Yours retrieves a NoMethodError
![image](https://user-images.githubusercontent.com/52209067/77689843-00449d00-6f68-11ea-87a8-da469331d92a.png)

- Real any?
![image](https://user-images.githubusercontent.com/52209067/77689883-10f51300-6f68-11ea-8a4c-319977bb3842.png)